powersoftau/src/utils.rs

```diff
@@ -49,27 +49,25 @@ pub fn print_hash(hash: &[u8]) {
 }
 
 /// Exponentiate a large number of points, with an optional coefficient to be applied to the
-/// exponent. Panics if provided arguments have different length.
+/// exponent.
 pub(crate) fn batch_exp<C: AffineCurve>(
     bases: &mut [C],
     exps: &[C::ScalarField],
     coeff: Option<&C::ScalarField>,
 ) {
-    // ensure the 2 vectors have the same length
-    assert_eq!(bases.len(), exps.len());
```

Comment:
why shouldn't we do this?

Reply:
or something similar

Reply:
We now use [Zip](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip) which will just short circuit if any of the two iterators returns None. 

E.g. if we gave 10 base with length 10 and 9 exponents, it'd exponentiate only the first 9 bases. Similarly, if we gave 10 exponents and 9 bases, it'd just ignore the exponents after the 9th

Reply:
yeah, I thought that's what it does, and that seems quite dangerous to me!